python/tests/conftest.py

```diff
@@ -202,6 +218,29 @@ def _version_store_factory_impl(
     return out
 
 
+def lmdb_version_store_cleanup(version_store_fixture):
+
+    @functools.wraps(version_store_fixture)
+    def wrapped(*args, **kwargs):
+        result: NativeVersionStore = version_store_fixture(*args, **kwargs)
+
+        yield result
+
+        #  pytest holds a member variable `cached_result` equal to `result` above which keeps the storage alive and
```

Comment:
This was a nightmare to figure out

Reply:
We should be able to award medals for valour above and beyond the call of duty.